Bluetooth: Audio: Fix issues the unicast client creating groups before configuring streams

Thalley opened the pull request:
This PR fixes an issue with releasing streams, as well as an issue that prohibited the unicast client from creating the unicast group before configuring the streams. 

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Fix issue with deleting unicast groups

There was a bug in bt_audio_iso_unbind_ep that caused an
assert, missing unbinding on stream released callback
and missing state check in bt_audio_unicast_group_delete

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add (un)bind of audio iso for streams

This allows us to allocate and bind the Audio ISO structs
to Audio Streams, thus allowing us to create the unicast group
before they have been configured.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add AUDIO_ISO debug logging

Add a logging module for the audio_iso.c file, to enable
logging for that.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

MariuszSkamra commented on subsys/bluetooth/audio/audio_iso.c:
```diff
@@ -242,3 +234,72 @@ struct bt_audio_ep *bt_audio_iso_get_ep(bool unicast_client,
+void bt_audio_iso_bind_stream(struct bt_audio_iso *audio_iso,
+			      struct bt_audio_stream *stream)
+{
+	struct bt_audio_iso_dir *audio_iso_ep;
```
s/audio_iso_ep/audio_iso_dir

MariuszSkamra commented on subsys/bluetooth/audio/audio_iso.c:
```diff
@@ -242,3 +234,72 @@ struct bt_audio_ep *bt_audio_iso_get_ep(bool unicast_client,
+void bt_audio_iso_unbind_stream(struct bt_audio_iso *audio_iso,
+				struct bt_audio_stream *stream)
+{
+	struct bt_audio_iso_dir *audio_iso_ep;
```
ditto

MariuszSkamra reviewed (dismissed):
LGTM

Thalley commented:
Fixed bad check for `if (ep->iso == NULL) {` which should have been (and now is) `if (stream->audio_iso == NULL) {` in `bt_audio_stream_qos`

Thalley committed:
Bluetooth: Audio: Fix issue with unicast client src/snk ASE count

If either the sink or source ASE count was zero, calls to
ARRAY_SIZE(srcs) or ARRAY_SIZE(sinks) would cause a build warning.

The arrays should actually not be there at all if the respective
ASE count was 0, as that is just a waste of memory. The arrays,
and all uses of them, have been properly guarded.

This also adds a build assert to ensure that at least one
of them is non-zero, and that we also test building either
of them with the value 0.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Fix issue with creating unidirectional CIS

As the unicast client, we would always create bidirectional
CIS to ensure that the PHY parameter is correctly set.
We can, however, just set the (required) PHY value and
leave the SDU and RTN values as 0, to avoid
creating bidirectional CIS.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Added 2 commits to fix a build warning and revert a bad workaround to handle missing PHY values.

MariuszSkamra commented:
@sjanc Could you take a look at this one?
